optuna/storages/_journal/base.py

```diff
@@ -0,0 +1,23 @@
+import abc
+from typing import Any
+from typing import Dict
+from typing import List
+
+
+class BaseJournalLogStorage(abc.ABC):
+    """Base class for Journal storages.
+
+    Storage classes implementing this base class must guarantee process safety. Thread-safety
+    need not be guaranteed. Use :class:`~optuna.storages.JournalFileSymlinkLock` or
```

Comment:
I'm not sure that it's appropriate to mention `JournalFileSymlinkLock` and `JournalFileOpenLock` here as those classes seem too specific to `JournalFileStorage`. For example, let's assume that an implemention of this abstract class uses RDB (or Redis, etc) as the storage and users want to run distributed optimizations from multiple machines, `JournalFileSymlinkLock` and `JournalFileOpenLock` could not (and should not) be used for such purposes (instead there would be RDB specific locking mechanisms we could use).

Reply:
My intention of this doc-string was to mention those `Journal...Locks` as an option for storages which do not internally have locking or multi-process exclusion mechanisms. That's why I wrote `Use ... if your storage
    does not natively support multi-process query executions.`. I'll fix the doc-string because it seems ambiguous.

Reply:
Fixed in https://github.com/optuna/optuna/pull/3854/commits/948684ae8fde0f0da4ee2d508dac383a360005ec.